src/GrooveApiClient/GrooveClient.cs

```diff
@@ -1,566 +1,567 @@
+                headers => GetAsync<UserProfileResponse>(
+                    Hostname,
+                    $"/1/user/{mediaNamespace}/profile",
+                    new CancellationToken(false),
+                    requestParameters,
+                    headers));
+        }
+        #endregion
+
+        private async Task<TResponse> ApiCallWithUserAuthorizationHeaderRefreshAsync<TResponse>(
+            Func<Dictionary<string, string>, Task<TResponse>> apiCall)
+            where TResponse : BaseResponse
+        {
+            string userAuthorizationHeader = await _userTokenManager.GetUserAuthorizationHeaderAsync();
+            if (string.IsNullOrEmpty(userAuthorizationHeader))
+            {
+                throw new ArgumentNullException("UserAuthorizationHeader", "Couldn't obtain a user authorization header");
+            }
+
+            Dictionary<string, string> requestHeaders = await FormatRequestHeadersAsync(userAuthorizationHeader);
+
+            TResponse response = await apiCall(requestHeaders);
+
+            if (response.Error?.ErrorCode == ErrorCode.INVALID_AUTHORIZATION_HEADER.ToString("G"))
+            {
+                userAuthorizationHeader = await _userTokenManager.GetUserAuthorizationHeaderAsync(true);
+                if (string.IsNullOrEmpty(userAuthorizationHeader))
+                {
+                    throw new ArgumentNullException("UserAuthorizationHeader", "Couldn't obtain a user authorization header");
+                }
+
+                requestHeaders = await FormatRequestHeadersAsync(userAuthorizationHeader);
+                response = await apiCall(requestHeaders);
+            }
+
+            return response;
+        }
+
+        private async Task<Dictionary<string, string>> FormatRequestParametersAsync(
+            string continuationToken = null,
+            string language = null,
+            string country = null,
+            ContentSource? source = null)
+        {
+            Dictionary<string, string> requestParameters = new Dictionary<string, string>();
+
+            if (!string.IsNullOrEmpty(continuationToken))
+                requestParameters.Add("continuationToken", continuationToken);
+
+            if (!string.IsNullOrEmpty(language))
+                requestParameters.Add("language", language);
+
+            if (!string.IsNullOrEmpty(country))
+                requestParameters.Add("country", country);
+
+            if (source.HasValue)
+            {
+                string sources = source.ToString().Replace(", ", "+");
+                requestParameters.Add("source", sources);
+            }
+
+            return requestParameters;
+        }
+
+        private async Task<Dictionary<string, string>> FormatRequestHeadersAsync(string userAuthorizationHeader)
+        {
+            MicrosoftAccountAuthenticationCache.AccessToken token =
+                await _microsoftAccountAuthenticationCache.CheckAndRenewTokenAsync(new CancellationToken(false));
+

```

Comment:
This version of the sdk does not support user auth

---
In reply to: [94612088](https://github.com/Microsoft/groove-api-sdk-csharp/pull/1#discussion_r94612088) [](ancestors = 94612088)